getndazn/dazn-lambda-powertools
# Feature/lambda layer

## What did you implement:

Closes #90 #68 



## How did you implement it:



Publish the layer through a SAR app using [this approach](https://serverless.pub/sar-layers/) which allows us to communicate the semantic version of the layer to the user.

Who can then reference the actual version from the SAR app's outputs.

To get the automation working, I 
* added a `build.sh` script to install every project in this `layer/nodejs` folder
* this folder is then zipped and shipped as the layer
* the structure of the `layer` folder (that it has a `nodejs` folder) maps to the NODE_PATH in the lambda environment, so when someone does `require('@dazn/xxx')` the runtime would check in our `nodejs` folder and find the powertools modules

## How can we verify it:



The SAR app is already public for v1.11.0 (the latest version in `lerna.json`): https://console.aws.amazon.com/lambda/home?region=us-east-1#/create/app?applicationId=arn:aws:serverlessrepo:us-east-1:374852340823:applications/dazn-lambda-powertools

If you deploy this to your account, then you will find the layer in your region
![image](https://user-images.githubusercontent.com/546969/64465382-b354c280-d10c-11e9-816b-7fadb822d3da.png)

Create a new folder, and add the layer
![image](https://user-images.githubusercontent.com/546969/64465414-d1222780-d10c-11e9-8454-29dc28d05607.png)

In the function code, reference one of the powertools
![image](https://user-images.githubusercontent.com/546969/64465427-e5febb00-d10c-11e9-8ed4-38d6099271ab.png)

Save and run and see that it worked even though we didn't package the logger with the package
![image](https://user-images.githubusercontent.com/546969/64465451-03338980-d10d-11e9-8522-06aa1b6052d5.png)

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
